components/automate-ui/src/app/modules/infra-proxy/infra-role-details/infra-role-details.component.ts

```diff
@@ -0,0 +1,216 @@
+import { Component, OnInit, OnDestroy, EventEmitter, ViewChild } from '@angular/core';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { combineLatest, Subject } from 'rxjs';
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+import { LayoutFacadeService, Sidebar } from 'app/entities/layout/layout.facade';
+import { routeParams, routeURL } from 'app/route.selectors';
+import { filter, pluck, takeUntil } from 'rxjs/operators';
+import { identity } from 'lodash/fp';
+import { infaRoleFromRoute } from 'app/entities/infra-roles/infra-role-details.selectors';
+import { GetRole } from 'app/entities/infra-roles/infra-role.action';
+import {
+  InfraRole, ExpandedList,
+  ChildList, List, RoleAttributes
+} from 'app/entities/infra-roles/infra-role.model';
+import { Node, Options } from '../treetable/models';
+import { JsonTreeableComponent as Treeable } from './../json-treeable/json-treeable.component';
+
+export type InfraRoleTabName = 'runList' | 'attributes';
+
+@Component({
+  selector: 'app-infra-role-details',
+  templateUrl: './infra-role-details.component.html',
+  styleUrls: ['./infra-role-details.component.scss']
+})
+
+export class InfraRoleDetailsComponent implements OnInit, OnDestroy {
+  public role: InfraRole;
+  public tabValue: InfraRoleTabName = 'runList';
+  public url: string;
+  public conflictErrorEvent = new EventEmitter<boolean>();
+  public modalType: string;
+  public serverId;
+  public OrgId;
+  public name;
+  public runList: string[];
+  public expandedList: ExpandedList[] = [];
+  public expandRunList: List[] = [];
+  public show = false;
+  public data: any = [];
+  public env_id = '_default';
+  public idList: any = [];
+  public hasRun_List = true;
+  public childNodes: Node<ChildList>[] = [];
+  private isDestroyed = new Subject<boolean>();
+  arrayOfNodesTree: Node<ChildList>[];
+  roleDetailsLoading = true;
+  treeOptions: Options<ChildList> = {
+    capitalisedHeader: true
+  };
+
+  public attributes: RoleAttributes = new RoleAttributes({
```

Comment:
There is no ambiguity on the type here; better to use type inference.
```suggestion
  public attributes = new RoleAttributes({
```